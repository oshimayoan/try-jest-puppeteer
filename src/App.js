import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  state = {
    name: '',
    email: '',
    password: '',
    isSubmitted: false,
  };

  render() {
    let {name, email, password, isSubmitted} = this.state;
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <div>
          <input
            autoFocus
            type="text"
            placeholder="Name"
            onChange={e => this.setState({name: e.target.value})}
            data-test="name"
          />
        </div>
        <div>
          <input
            type="email"
            placeholder="Email"
            onChange={e => this.setState({email: e.target.value})}
            data-test="email"
          />
        </div>
        <div>
          <input
            type="password"
            placeholder="Password"
            onChange={e => this.setState({password: e.target.value})}
            data-test="password"
          />
        </div>
        <div>
          <button
            onClick={() => {
              if (name && email && password) {
                this.setState({isSubmitted: true});
              }
            }}
            data-test="submitButton"
          >
            Submit
          </button>
        </div>
        {isSubmitted && <p data-test="submitSuccess">Submitted!</p>}
      </div>
    );
  }
}

export default App;
