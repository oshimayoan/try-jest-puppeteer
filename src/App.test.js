import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<App />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
