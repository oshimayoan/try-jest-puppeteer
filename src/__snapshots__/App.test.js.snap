// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="mock-file-jest"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <div>
    <input
      autoFocus={true}
      data-test="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
    />
  </div>
  <div>
    <input
      data-test="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
  </div>
  <div>
    <input
      data-test="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
  </div>
  <div>
    <button
      data-test="submitButton"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;
